.github/workflows/dev_container.yml

```diff
@@ -0,0 +1,25 @@
+name: Check research template dev container functions
+
+on:
+  schedule:
+    - cron: "18 8 * * *"
+  workflow_dispatch:
+
+jobs:
+  dev-container:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout research-template repository
+        uses: actions/checkout@v4
+        with:
+          repository: 'opensafely/research-template'
```

Comment:
Checking out and running opensafely/research-template seems like unnecessarily tight coupling to me. What are the advantages over, say, checking in a simple `dataset_definition.py` to this repo and calling `opensafely exec`?

Reply:
The goal of this test is to check end-to-end that the dev container setup in `research-template` runs (which is a little indirect, I agree).

However, it was suggested [originally to keep this tests here](https://github.com/opensafely/research-template/pull/123#issuecomment-2090166248) rather than in the `research-template` repository.

The reason to checkout the repository is so that we build a dev container from the `.devcontainer` configuration there.

That said, I'm not sure if it's necessary to run the full research-template project (which is also tested when on PRs to that repository).

The only real extra pieces of installation that happen in the research-template `.devcontainer` configuration are:

* the OpenSAFELY CLI
* ehrQL for autocompletion

So maybe it's sufficient to check those are installed correctly (by running and importing respectively).

(It's also possible that those could move in future to here.)

Reply:
I'm open to this being moved or restructured, but maybe that's the subject of a subsequent PR.